Radar plot

soaubier opened the pull request.

soaubier committed:
first radar plot add

soaubier committed:
Clean up: remove print(), improve variable names ...

Djedouas commented:
Threshold spinbox maximum value must be a lot greater than 99.

If the threshold checkbox is not checked, we should hide the threshold widgets (line type and value), no?

Threshold checkbox need a first upper letter.

Djedouas commented on DataPlotly/core/plot_factory.py:
```diff
@@ -740,9 +758,9 @@ def build_figures(self, plot_type, ptrace, config=None) -> str:
-
         else:
             figures = go.Figure(data=ptrace, layout=self.layout)
+        print(ptrace)
```
```suggestion
```

soaubier committed:
Set max threshold value & adjust QDoubleSpinBox size

soaubier committed:
add radar icons

soaubier committed:
Load configuration settings added

soaubier committed:
Add font style management

soaubier committed:
Adjust widget size for threshold

soaubier committed:
manage CLI radar option

soaubier committed:
fixup! Load configuration settings added

Djedouas committed:
minor refacto for a better y_radar_values management

Djedouas committed:
correct default settings

Djedouas commented on DataPlotly/core/plot_types/icons/radar.svg:
I think we need something more recognizable: not filled and with more than 3 points.

See this example

![image](https://github.com/user-attachments/assets/d20b301d-4769-40c0-855b-ffacf0e1f1bd)

Djedouas commented on DataPlotly/core/plot_types/radar.py:
```diff
@@ -0,0 +1,108 @@
+        # Sample colors from the color scale based on the length of settings.y
+        colors_list = pc.sample_colorscale(settings.properties['color_scale'], np.linspace(0, 1, len(settings.y_radar_values[0])))
+        # List repeating the line type for each element in settings.y
+        line_type_list = [settings.properties['line_dash']] * len(settings.y_radar_values)
```
This is the same line_type for every trace, we don't need to have a list.

We can use the line type in the `for` loop.

Djedouas replied:
In fact we need it, because if threshold is used, we add another line type at the end of the line_type_list

Djedouas commented on DataPlotly/core/plot_types/radar.py:
```diff
@@ -0,0 +1,108 @@
+        line_type_list = [settings.properties['line_dash']] * len(settings.y_radar_values)
+
+        # Add a black color and a threshold line to the data 
+        if settings.properties['threshold'] is True :
```
```suggestion
        if settings.properties['threshold']:
```

Djedouas commented on DataPlotly/core/plot_types/radar.py:
```diff
@@ -0,0 +1,108 @@
+        if settings.properties['threshold'] is True :
+            colors_list.append('#000000')
+            settings.y.append([settings.properties['threshold_value']] * len(settings.y_radar_values[0]))
+            settings.y_radar_labels.append('threshold')
```
We might want to translate this label

Djedouas commented on DataPlotly/core/plot_types/radar.py:
```diff
@@ -0,0 +1,108 @@
+            colors_list.append('#000000')
+            settings.y.append([settings.properties['threshold_value']] * len(settings.y_radar_values[0]))
+            settings.y_radar_labels.append('threshold')
+            line_type_list.append(settings.properties['line_type_threshold'])
```
cf. previous comment

Djedouas replied:
cf previous answer

Djedouas commented on DataPlotly/gui/plot_settings_widget.py:
```diff
@@ -459,7 +458,7 @@ def data_defined_color_updated(self):
         if self.in_color_defined_button.isActive():
             # if plot is type for which using an expression for the color selection makes sense
-            if self.ptype in ['scatter', 'bar', 'pie', 'ternary', 'histogram']:
+            if self.ptype in ['scatter', 'bar', 'pie', 'ternary', 'histogram', 'radar']:
```
Do we really need to add radar plot here?

soaubier replied:
```suggestion
            if self.ptype in ['scatter', 'bar', 'pie', 'ternary', 'histogram']:
```

Djedouas commented on DataPlotly/gui/plot_settings_widget.py:
```diff
@@ -670,8 +672,8 @@ def refreshWidgets(self):  # pylint: disable=too-many-statements,too-many-branch
 
         # ScatterPlot marker types
         self.marker_types = OrderedDict([
-            (self.tr('Points'), 'markers'),
             (self.tr('Lines'), 'lines'),
+            (self.tr('Points'), 'markers'),
             (self.tr('Points and Lines'), 'lines+markers')
```
Why this change of order?

Djedouas commented on DataPlotly/gui/plot_settings_widget.py:
```diff
@@ -724,8 +726,10 @@ def refreshWidgets(self):  # pylint: disable=too-many-statements,too-many-branch
         self.line_combo.clear()
-        for k, v in self.line_types.items():
+        self.line_type_threshold.clear()
+        for k, v in self.line_types2.items():
```
Why changing from `line_types` to `line_types2`?

Djedouas committed:
fix bad variable usage

soaubier committed:
Update DataPlotly/core/plot_types/radar.py

Co-authored-by: Jacky Volpes <34267385+Djedouas@users.noreply.github.com>

soaubier committed:
Update DataPlotly/gui/plot_settings_widget.py

soaubier committed:
review

soaubier committed:
add traduction to radar.py

soaubier committed:
radar svg update

Djedouas committed:
fix default settings for line_type_threshold

Djedouas committed:
re-add feature_subset_defined option for radar plot

Djedouas committed:
fix missing import

Djedouas committed:
fix radar threshold

line type and combo box are now decorrelated

Djedouas committed:
remove duplicated line

Djedouas committed:
fix widgets synchronization

Refresh the UI according to Points, Line, Points and Lines marker type
not only when marker type changes, but also when the plot type changes
(to be correctly initialized).

Djedouas committed:
add default values for radar threshold
